windows/cross-compile/build.sh

```diff
@@ -92,1399 +76,45 @@ then
-" > toolchain.cmake
-
-# Run tests using Wine
-set +u
-if [[ -n "$TRAVIS_CI_STAGE" ]]
-then
-  echo "SET(TEST_CROSSCOMPILING_EMULATOR /usr/bin/wine)" >> toolchain.cmake
-fi
-set -u
-
 # Spell check on windows currently not supported, disable
 if [[ "$BUILD_TYPE" == "release" ]]
 then
-  cmake -DCMAKE_TOOLCHAIN_FILE=./toolchain.cmake \
+  cmake -DCMAKE_TOOLCHAIN_FILE=/build/windows-toolchain.cmake \
+        -DCMAKE_PREFIX_PATH=/windows \
         -DCMAKE_BUILD_TYPE=Release \
         -DSPELL_CHECK=OFF \
         -DDESKTOP_NOTIFICATIONS=ON \
         -DUPDATE_CHECK=ON \
         -DSTRICT_OPTIONS=ON \
-        ..
+        "$QTOX_SRC_DIR"
 elif [[ "$BUILD_TYPE" == "debug" ]]
 then
-  cmake -DCMAKE_TOOLCHAIN_FILE=./toolchain.cmake \
+  cmake -DCMAKE_TOOLCHAIN_FILE=/build/windows-toolchain.cmake \
+        -DCMAKE_PREFIX_PATH=/windows \
         -DCMAKE_BUILD_TYPE=Debug \
         -DSPELL_CHECK=OFF \
         -DDESKTOP_NOTIFICATIONS=ON \
         -DUPDATE_CHECK=ON \
         -DSTRICT_OPTIONS=ON \
-        ..
+        "$QTOX_SRC_DIR"
 fi
 
-make
+make -j $(nproc)
 
+mkdir -p "${QTOX_PREFIX_DIR}"
 cp qtox.exe $QTOX_PREFIX_DIR
-cp $QT_PREFIX_DIR/bin/Qt5Core.dll \
-   $QT_PREFIX_DIR/bin/Qt5Gui.dll \
-   $QT_PREFIX_DIR/bin/Qt5Network.dll \
-   $QT_PREFIX_DIR/bin/Qt5Svg.dll \
-   $QT_PREFIX_DIR/bin/Qt5Widgets.dll \
-   $QT_PREFIX_DIR/bin/Qt5Xml.dll \
-   $QTOX_PREFIX_DIR
-cp -r $QT_PREFIX_DIR/plugins/imageformats \
-      $QT_PREFIX_DIR/plugins/platforms \
-      $QT_PREFIX_DIR/plugins/iconengines \
-      $QTOX_PREFIX_DIR
-cp {$OPENSSL_PREFIX_DIR,$SQLCIPHER_PREFIX_DIR,$FFMPEG_PREFIX_DIR,$OPENAL_PREFIX_DIR,$QRENCODE_PREFIX_DIR,$EXIF_PREFIX_DIR,$OPUS_PREFIX_DIR,$SODIUM_PREFIX_DIR,$VPX_PREFIX_DIR,$TOXCORE_PREFIX_DIR}/bin/*.dll $QTOX_PREFIX_DIR
-
-cp "$SNORE_PREFIX_DIR/bin/libsnore-qt5.dll" $QTOX_PREFIX_DIR
-mkdir -p "$QTOX_PREFIX_DIR/libsnore-qt5"
-cp "$SNORE_PREFIX_DIR/lib/plugins/libsnore-qt5/libsnore_backend_windowstoast.dll" "$QTOX_PREFIX_DIR/libsnore-qt5"
-cp "$SNORE_PREFIX_DIR/bin/SnoreToast.exe" $QTOX_PREFIX_DIR
-
-cp /usr/lib/gcc/$ARCH-w64-mingw32/*-posix/{libgcc_s_*.dll,libstdc++*.dll,libssp*.dll} $QTOX_PREFIX_DIR
-cp /usr/$ARCH-w64-mingw32/lib/libwinpthread*.dll $QTOX_PREFIX_DIR
-
-# Setup wine
-# Note that SQLCipher and FFmpeg (maybe more?) seem to setup ~/.wine on their
-# own, but to the wrong bitness (always 64-bit?), when we want a matching
-# bitness here for mingw-ldd, so remove it before proceeding.
-rm -rf ~/.wine
-if [[ "$ARCH" == "i686" ]]
-then
-  export WINEARCH=win32
-elif [[ "$ARCH" == "x86_64" ]]
-then
-  export WINEARCH=win64
-fi
-winecfg
+cp -r /export/* $QTOX_PREFIX_DIR
 
 # Run tests (only on Travis)
 set +u
 if [[ -n "$TRAVIS_CI_STAGE" ]]
 then
-  # Add libgcc_s_*.dll, libwinpthread-1.dll, QtTest.dll, etc. into PATH env var of wine
-  export WINEPATH=`cd $QTOX_PREFIX_DIR ; winepath -w $(pwd)`\;`winepath -w $QT_PREFIX_DIR/bin/`
+  export WINEPATH='/export;/windows/bin'
```

Comment:
Why did you remove the use of `winepath`? You want to use `winepath` in here, because these should be Windows paths accessible from within Wine, e.g.t  they should use the Windows backslash \ folder delimiter instead of forward slash / directory delimiter of Linux, use the Windows drive letter, e.g. `C:\`, etc.

Reply:
Wine supports both. I'm not sure what you mean by "should" here. I think converting the paths to windows paths for wine to convert them back to unix paths is a little roundabout when we can just give wine the unix paths directly. Maybe wine will do this double conversion under the hood again, but I think this is clearer.

Reply:
It doesn't convert them, `WINEPATH` is prepended to the `PATH` variable inside wine as it is.

```
$ mkdir /tmp/test
$ touch /tmp/test/foobar
$ WINEPATH=/tmp/test wine cmd
Microsoft Windows 6.1.7601

Z:\home\nurupo>echo %PATH%
/tmp/test;C:\windows\system32;C:\windows;C:\windows\system32\wbem

Z:\home\nurupo>dir /tmp/test
Invalid parameter.

Z:\home\nurupo>
```

See, `PATH` contains `/tmp`, which Windows programs wouldn't understand.

You have to convert it using the `winepath` program to a Windows path, e.g.:

```bash
$ mkdir /tmp/test
$ touch /tmp/test/foobar
$ WINEPATH="$(winepath -w /tmp)" wine cmd
Microsoft Windows 6.1.7601

Z:\home\nurupo>echo %PATH%
Z:\tmp\test;C:\windows\system32;C:\windows;C:\windows\system32\wbem

Z:\home\nurupo>dir Z:\tmp\test

Directory of Z:\tmp\test

 1/10/2022   10:33 AM             0  foobar
       1 file                         0 bytes
       0 directories    100,974,759,936 bytes free

Z:\home\nurupo>
```

I have no idea of what you mean by "Wine supports both". It clearly does not.